distributed/scheduler.py

```diff
@@ -506,6 +506,12 @@ def del_scheduler_file():
 
         self.loop.add_callback(self.reevaluate_occupancy)
 
+        def start_pcs():
```

Comment:
Perhaps this infrastructure should be in distributed/core.py?

Reply:
I've moved this to core.py and applied that logic consistently across, scheduler.py, worker.py, and nanny.py